src/state/__tests__/state-machine/expansion/hazardous-substance-regulation.spec.ts

```diff
@@ -0,0 +1,172 @@
+import { expect } from "vitest";
+import { getTestActor, testRandomSeed } from "@/state/__tests__/utils";
+import { concat, filter, first, without } from "lodash";
+import i18n from "@/i18n";
+
+testRandomSeed("having shared animal and plant in hand unlocks habitat", async (seed) => {
+  const { send, getState, activatePolicy } = getTestActor({
+    useSpecialCards: true,
+    playerCount: 1,
+    seed,
+  });
+  const stateBefore = getState();
+
+  stateBefore.plantMarket.deck = concat(stateBefore.plantMarket.deck, stateBefore.plantMarket.table);
+  stateBefore.plantMarket.table = [];
+  stateBefore.animalMarket.deck = concat(stateBefore.animalMarket.deck, stateBefore.animalMarket.table);
+  stateBefore.animalMarket.table = [];
+
+  const plantCard = first(filter(stateBefore.plantMarket.deck, (card) => card.habitats.includes("pelagic")))!;
+  const animalCard = first(filter(stateBefore.animalMarket.deck, (card) => card.habitats.includes("pelagic")))!;
+
+  stateBefore.players[0].hand.push(plantCard);
+  stateBefore.players[0].hand.push(animalCard);
+  stateBefore.plantMarket.deck = without(stateBefore.plantMarket.deck, plantCard);
+  stateBefore.animalMarket.deck = without(stateBefore.animalMarket.deck, animalCard);
+
+  activatePolicy({
+    policyName: "Hazardous substance regulation",
+    stateBefore,
+    specialCardSource: "plants",
+  });
+  send({ type: "user.click.player.hand.card", card: plantCard });
+  send({ type: "user.click.player.hand.card", card: animalCard });
+
+  const state = getState();
+  expect(state.commandBar).toBeUndefined();
+  expect(
+    state.habitatMarket.deck
+      .filter((habitatTile) => habitatTile.isAcquired)
+      .every(
+        (habitatTile) =>
+          habitatTile.name === "baltic" ||
+          (plantCard.habitats.includes(habitatTile.name) && animalCard.habitats.includes(habitatTile.name)),
+      ),
+  );
+});
+
+testRandomSeed("cant select animal that doesn't share habitat", async (seed) => {
+  const { send, getState, activatePolicy } = getTestActor({ useSpecialCards: true, playerCount: 1, seed });
+  const stateBefore = getState();
+
+  stateBefore.plantMarket.deck = concat(stateBefore.plantMarket.deck, stateBefore.plantMarket.table);
+  stateBefore.plantMarket.table = [];
+  stateBefore.animalMarket.deck = concat(stateBefore.animalMarket.deck, stateBefore.animalMarket.table);
+  stateBefore.animalMarket.table = [];
+
+  const plantCard = first(stateBefore.plantMarket.deck)!;
+  const animalCard = first(
+    filter(stateBefore.animalMarket.deck, (card) =>
+      plantCard?.habitats.every((habitat) => !card.habitats.includes(habitat)),
+    ),
+  )!;
+  stateBefore.players[0].hand.push(plantCard);
+  stateBefore.players[0].hand.push(animalCard);
+  stateBefore.plantMarket.deck = without(stateBefore.plantMarket.deck, plantCard);
+  stateBefore.animalMarket.deck = without(stateBefore.animalMarket.deck, animalCard);
```

Comment:
It all seems a bit overcomplicated

```suggestion
  const plantCard = removeOne(stateBefore.plantMarket.deck, ({ habitats }) => habitats.includes("rock"))!;
  const animalCard = removeOne(
    stateBefore.animalMarket.deck,
    ({ habitats }) => intersection(habitats, plantCard.habitats).length === 0,
  )!;
  
  stateBefore.players[0].hand.push(plantCard);
  stateBefore.players[0].hand.push(animalCard);
```

Reply:
This was done for a specific reason, because sometimes the game is seeded in such a way that many of the special cards are on the table and not in the deck, causing special cards to not be found. But I agree that it can be simplified a bit, this is kinda copy pasta, I will see what I can do

Reply:
Just checked the spreadsheet. There are 5 animals and 5 plants with special ability. So there is always at least one on the table